mypy/main.py

```diff
@@ -1018,6 +1021,18 @@ def set_strict_flags() -> None:
     if options.logical_deps:
         options.cache_fine_grained = True
 
+    # Determine if mypy is installed in a non-default prefix
+    try:
+        mypy_sitepkgs_path = pathlib.Path(__file__).parents[1]  # <sitepkgs>/mypy/main.py
+    except IndexError:
+        pass
+    else:
+        if (
+            mypy_sitepkgs_path.name == "site-packages"
```

Comment:
Sometimes on e.g. debian systems things are installed into `dist-packages`, and in general the prefix can be any directory on the path, as far as I am aware, so I think this check is wrong. Same as the code below, I think it is perhaps not general enough.

Reply:
There is a difference between prefix and [target](https://pip.pypa.io/en/stable/cli/pip_install/#cmdoption-t). A library installed in a prefix will always end up in a site-packages subfolder, so I believe this check is correct as it's not trying to support target (nor OS-specific) paths.

Reply:
Is that documented somewhere? Even without prefix, packages installed via pip end up in folders not called site-packages. Also the prefix doesn't always look like /lib/(pythonX.Y)/ (e.g. on Mac there is /Library/Python/3.8/site-packages).

Reply:
Well this is how Python derives the site-packages path from a prefix https://github.com/python/cpython/blob/8187033de10ee94c4997ab619f9ee4f255aa0b4c/Lib/site.py#L319

Reply:
Debian and derivatives patch site.py: "dist-packages instead of site-packages. Third party Python software installed from Debian packages goes into dist-packages, not site-packages. This is to reduce conflict between the system Python, and any from-source Python build you might install manually. "

Reply:
> Debian and derivatives patch site.py: "dist-packages instead of site-packages. 

Yes I do get this, but I don't think `mypy` should have any knowledge of OS-specific details. Python uses `site-packages` officially so I think this is what `mypy` should support IMHO.

Reply:
While I wish we could ignore Debian patches to Python, a lot of mypy core devs use Debian based distros personally (like me) or at work (e..g. Dropbox, from what I understand), and Ubuntu/Debian remain very popular. It isn't an option to ignore them, it would be a significant breaking change and cause pain for people just starting to use mypy.

Fortunately, I read through what pip actually does when you specify a prefix, it seems that it uses [whatever distutils says](https://github.com/pypa/pip/blob/37a2b12a21450ed03c886e16bb454464783ef687/src/pip/_internal/locations/_distutils.py#L21), so perhaps we could mirror that logic?

Reply:
Right, I've had another think about this and removed the check `mypy_sitepkgs_path.name == "site-packages"`. This is the reasoning: if `mypy` is installed in `dist-packages` then it will have a default prefix, which we can easily check against `sitepkgs.getsitepackages()`. If it's not a default prefix, then it must be a "`site-packages`" prefix, at which point we can just apply `site`'s logic to infer just the actual prefix.